src/poc/miner_poc_grpc_client_statem.erl

```diff
@@ -1151,6 +1162,57 @@ handle_stability_check(CurState, Data = #data{down_events_in_period = NumDownEve
             {keep_state, Data#data{down_events_in_period = 0, stability_check_timer = TRef}}
     end.
 
+handle_block_age_check(_CurState, #data{connection = undefined} = Data) ->
+    %% your request for block age cannot be completed at this time; please try again later
+    TRef = erlang:send_after(?BLOCK_AGE_TIMEOUT, self(), block_age_check),
+    {keep_state, Data#data{block_age_timer = TRef}};
+handle_block_age_check(CurState, #data{connection = Connection} = Data) ->
+    case send_block_age_req(Connection) of
+        {ok, BlockAge} when BlockAge >= ?MAX_BLOCK_AGE ->
+            %% we're bailing from this validator so reset the down events to start fresh
+            Data1 = Data#data{down_events_in_period = 0},
+            lager:warning("validator block age ~p exceeded maximum allowed; switching validators", [BlockAge]),
+            case CurState of
+                setup -> {repeat_state, Data1};
+                _ -> {next_state, setup, Data1}
+            end;
+        {error, _Reason} ->
+            %% validator, you had one job...
+            Data1 = Data#data{down_events_in_period = 0},
+            lager:warning("validator block age request failed; switching validators", [_Reason]),
+            case CurState of
+                setup -> {repeat_state, Data1};
+                _ -> {next_state, setup, Data1}
+            end;
+        _ ->
+            %% connected validator seems to be keeping up; once more around the loop
+            TRef = erlang:send_after(?BLOCK_AGE_TIMEOUT, self(), block_age_check),
+            {keep_state, Data#data{block_age_timer = TRef}}
+    end.
+
+send_block_age_req(Connection) ->
+    try
+        lager:info("Send block age unary request", []),
```

Comment:
Is this going to be really spammy? Maybe it should be `debug`

Reply:
quite possibly; we have a lot of those at the moment where we've dialed up the logging. although if this is unlikely to make a difference having the additional logging might not be valuable
for now i was just mirroring the log level of the statements in the `send_grpc_unary_req` function

Reply:
looking again, i don't see a ton of value in this message being at the `info` level other than as a "i'm still alive doing what i'm doing" so i turned it down to debug